Cognite.Simulator.Utils/StateUtils.cs

```diff
@@ -47,7 +48,10 @@ await store.DeleteExtractionState(states, tableName, token)
                 .ConfigureAwait(false);
             foreach (var state in states)
             {
-                DeleteLocalFile(state.FilePath);
+                // get directory path from file path 
+                // (file path is the directory path + file name)
+                var dirPath = Path.GetDirectoryName(state.FilePath);
```

Comment:
And this is the risky part. If you try to delete a FilePath that is using the previous format, you would actually ask it to delete the whole `files` folder.

Reply:
Yes, exactly what i meant as the worst case scenario in my previous comment 😅 . However im not sure of what approach to take, should we delete the state.db when we install a new version? or should we handle backward compatibility directly in the code?

Reply:
What comes to mind is (considering that GAP is not production yet, and you can manually clean up current GAP installations):
- Deal with compatibility in code, with seamless transition.
- Add a new state property, like `state.Directory`. Assign it just for new models that are downloaded to directories.
- Delete the directory only for models that have assigned `state.Directory`, else keep the previous delete logic.
IMO, this keeps the intention of deleting a folder explicit. Not sure if this holds in practice, would need to be tested 😅

Reply:
This new logic, is now committed, we now maintain a `IsInDirectory` flag in our state db, this flag is only set for files which are saved in directories, its false for everything that was previously stored :)